src/components/organisms/UserProfile/UpdateSpaceInfo.tsx

```diff
@@ -0,0 +1,185 @@
+import { SuiteAmenities } from "@/constants";
+import onBoardingService from "@/utils/apis/onboarding";
+import Alert from "@/utils/base/alerts";
+import { SpaceInfoSchema, UpdateSpaceInfoSchema } from "@/utils/schema/details";
+import { type InferSchema } from "@/utils/schema/helpers";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Title, Text, Input, Button, Select } from "@the_human_cipher/components-library";
+import { useState } from "react";
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+
+type Inputs = InferSchema<typeof UpdateSpaceInfoSchema>;
+
+type Name = keyof Inputs;
+
+/**
+ * totally useless codes, just wanted to do some ts magic
+ */
+type Label = Name extends `space_${infer A}` ? `Space ${Capitalize<A>}` : Name;
+
+type Field = {
+  name: Name;
+  label: LoosenString<Label>;
+  placeholder: LoosenString<`Enter ${Lowercase<Label>}`>;
+};
+
+const fields: Field[] = [
+  {
+    name: "space_name",
+    label: "Space Name",
+    placeholder: "Enter space name",
+  },
+  {
+    name: "space_address",
+    label: "Space Address",
+    placeholder: "Enter space address ",
+  },
+  {
+    name: "space_size",
+    label: "Space Size (sq. ft.)",
+    placeholder: "Enter space size ",
+  },
+];
+
+const UpdateSpaceInfo = ({ isEditMode, setIsEditMode, userProfile }: ProfileProps) => {
+  const { register, formState, handleSubmit, control, watch } = useForm<Inputs>({
+    resolver: zodResolver(UpdateSpaceInfoSchema),
+    mode: "onChange",
+    values: {
+      space_name: userProfile.space?.space_name as string,
```

Comment:
you are using ?. to access the property, it means if the property exists give me if not, return undefined, a good way to have done it would be to

```tsx
const someVariable = myObject?.myProperty?.isString ?? "" 
// so if the value should be undefined give us an empty string
```